README.md

```diff
@@ -0,0 +1,103 @@
+## Generate a new Maven project ##
+
+- Run below command from Command line tool
+
+```text
+mvn archetype:generate -DartifactId=selenium-learning -DgroupId=com.mytest -DarchetypeGroupId=org.apache.maven.archetypes -DarchetypeArtifactId=maven-archetype-quickstart -DarchetypeVersion=1.4
+```
+
+- Run below command to download Maven Dependencies, compile, and build project
+
+```text
+mvn clean package -e
+```
+OR,
+
+```text
+mvn clean install -e
+```
+**NOTE:** Use argument `-e` to console error messages while building the project 
+
+- Run below command to run unit tests against compile source code using available testing framework
+
+```text
+mvn test
+```
+
+## Add required dependency for Maven project ##
+
+- Update/Modify the deppendencies in `pom.xml`. Refer [Maven Central Repository](https://central.sonatype.com/) to know about the latest version of dependencies.
+
+1. Add Selenium Dependency
+
+```xml
+<dependency>
+      <groupId>org.seleniumhq.selenium</groupId>
+      <artifactId>selenium-java</artifactId>
+      <version>4.18.1</version>
+</dependency>
+```
+2. Add Dependendency to load and manage Browsers
+
+```xml
+<dependency>
+      <groupId>io.github.bonigarcia</groupId>
+      <artifactId>webdrivermanager</artifactId>
+      <version>5.7.0</version>
+</dependency>
+```
+3. Add dependency TestNG testing framework
+
+**NOTE:** Dependency Management - `test` scope is to inform `maven` tat this dependency is not required at standard runtime of the application but will be used while Testing purposes only
+
+```xml
+<dependency>
+      <groupId>org.testng</groupId>
+      <artifactId>testng</artifactId>
+      <version>7.9.0</version>
+      <scope>test</scope>
+</dependency>
+```
+
+4. Add below support dependencies to avoid runtime error `Failed to load class org.slf4j.impl.StaticLoggerBinder`
+
+```xml
+    <dependency>
+    <groupId>org.slf4j</groupId>
+    <artifactId>slf4j-api</artifactId>
+    <version>2.0.12</version>
+</dependency>
+
+<dependency>
+    <groupId>org.slf4j</groupId>
+    <artifactId>slf4j-log4j12</artifactId>
+    <version>2.0.12</version>
+</dependency>
+```
```

Comment:
**Optimization Issue**: Including both 'slf4j-api' and 'slf4j-log4j12' dependencies might introduce version conflicts or redundancy, especially if the project or other dependencies bring their own SLF4J bindings. <br> **Fix**: Ensure that only one binding is included in the project to avoid classpath issues. If 'slf4j-log4j12' is necessary, consider excluding 'slf4j-api' explicitly if it's brought transitively by other dependencies. <br> **Code Suggestion**: 
 ```
 +<!-- To avoid runtime error 'Failed to load class org.slf4j.impl.StaticLoggerBinder' ensure only one SLF4J binding is present -->
  <dependency>
  <groupId>org.slf4j</groupId>
  <artifactId>slf4j-api</artifactId>
  <version>2.0.12</version>
  </dependency>
  <dependency>
  <groupId>org.slf4j</groupId>
  <artifactId>slf4j-log4j12</artifactId>
  <version>2.0.12</version>
  </dependency>
 ```